Assets/Scripts/Managers/GameManager.cs

```diff
@@ -66,15 +66,31 @@ public class GameManager : MonoBehaviour
         public event Action Reset;
 
         /// <summary>
-        /// Action invoked on game data changed.
+        /// Action invoked when a level is completed.
         /// </summary>
-        public event Action GameDataChanged;
+        public event Action saveGame;
 
         /// <summary>
         /// Canvas to fade in/out when transitioning between scenes
         /// </summary>
         [SerializeField] private FadeEffects sceneTransitionCanvas;
 
+        [SerializeField] private List<string> levelNameList;
+
+        //<summary>
+        //the numbers of times Marshmallow dies in a level
+        //called by results manager and level select to display stats
+        //</summary>
+        public int thisLevelDeaths;
+        
+        //<summary>
+        //the time it took for player to beat a level
+        //called by results manager and level select to display stats
+        // in the form of hh:mm:ss
+        //</summary>
+        public string thisLevelTime;
```

Comment:
should these be serialized? they're currently serialized

could swap them with an autoproperty like `{ get; private set;}`